src/main/kotlin/cacophony/automata/NFAConstructor.kt

```diff
@@ -0,0 +1,135 @@
+package cacophony.automata
+
+import kotlin.collections.emptyMap
+import cacophony.utils.AlgebraicRegex
+
+fun buildNFAFromRegex(regex: AlgebraicRegex): NFA<Int> = buildSimpleNFAFromRegex(regex)
+
+// implementation relies on the fact that the initial state will always have number 0
```

Comment:
Please start comment with uppercase letter and end it with dot to stay consistent with other PRs.